src/transformState.js

```diff
@@ -5,7 +5,42 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const addProps = (inputState, dataObj) => {
+    for (const [key, value] of Object.entries(dataObj)) {
+      inputState[key] = value;
+    }
+  };
+
+  const removeProps = (inputState, dataArray) => {
+    for (const key of dataArray) {
+      delete inputState[key];
+    }
+  };
+
+  const clear = (inputState) => {
+    for (const key in inputState) {
+      delete inputState[key];
+    }
+  };
```

Comment:
You can move this functions outside function transformState.
Now these functions are created afresh every time the transformState function is called.